code/modules/mob/living/simple_animal/hostile/hostile.dm

```diff
@@ -235,31 +243,26 @@
 			. += A
 			continue
 
-
-
-/mob/living/simple_animal/hostile/proc/Found(atom/A)//This is here as a potential override to pick a specific target if available
+/* Returning True on this proc means
+	that only this target is listed and
+	nothing more is considered. */
+/mob/living/simple_animal/hostile/proc/Found(atom/A)
 	return
 
-/mob/living/simple_animal/hostile/proc/PickTarget(list/Targets)//Step 3, pick amongst the possible, attackable targets
-	if(target != null)//If we already have a target, but are told to pick again, calculate the lowest distance between all possible, and pick from the lowest distance targets
-		for(var/pos_targ in Targets)
-			var/atom/A = pos_targ
-			var/target_dist = get_dist(targets_from, target)
-			var/possible_target_distance = get_dist(targets_from, A)
-			if(target_dist < possible_target_distance)
-				Targets -= A
-	if(!Targets.len)//We didnt find nothin!
-		return
-	var/chosen_target = pick(Targets)//Pick the remaining targets (if any) at random
-	return chosen_target
-
-// Please do not add one-off mob AIs here, but override this function for your mob
-/mob/living/simple_animal/hostile/CanAttack(atom/the_target)//Can we actually attack a possible target?
-	if(isturf(the_target) || !the_target || the_target.type == /atom/movable/lighting_object) // bail out on invalids
+/* If Found() does not return TRUE then this will be
+	used to sort out what we can attack. Its encouraged
+	that you override this for your mob instead of making
+	niche conditions. In some cases using Found() will suffice.*/
+/mob/living/simple_animal/hostile/CanAttack(atom/the_target)
+	if(isturf(the_target) || !the_target || the_target.type == /atom/movable/lighting_object)
+		// bail out on invalids
 		return FALSE
 
 	if(ismob(the_target)) //Target is in godmode, ignore it.
 		var/mob/M = the_target
+		//This is in order to make mobs not instantly reaggro on mobs they lost patience on.
+		if(mob_ignore_list[M] > world.time)
```

Comment:
New list that empties itself if its more than 2. In testing this usually results in ranged monsters pausing for a moment before resuming fire.